DESIGN.md

```diff
@@ -0,0 +1,370 @@
+# Apollo Android design discussion
+
+These are some preliminary ideas for mapping GraphQL query results to Java, based on what we currently do for Swift in Apollo iOS.
+
+## Query classes
+
+I would expect us to generate classes for queries (and other operations). Instances of these classes represent a particular query, fully configured with the required variables.
```

Comment:
Agree, another example if someone uses Realm Db and decides to use generated classes as a schema for db. Realm implies for schema models to be inherited from RealmObject. So with interfaces approach they can manually extend from RealmObject and implement generated interfaces. It's not quite good example from architecture, but still there is such possibility.